From e271b8222aea1e386c853d22b43091ae4f729380 Mon Sep 17 00:00:00 2001
From: Damodar Reddy GangiReddy <quic_dgangire@quicinc.com>
Date: Tue, 28 Jan 2025 15:41:00 +0530
Subject: [PATCH] Use system bus instead of session for obexd

Upstream now has enabled support for system bus using command line
option. In QCLINUX as session bus is not supported use system bus
instead of session bus.

Signed-off-by: Damodar Reddy GangiReddy <quic_dgangire@quicinc.com>
Upstream-Status: Inappropriate [target specific change]
---
 obexd/src/obex.service.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/obexd/src/obex.service.in b/obexd/src/obex.service.in
index 57f8b1418..b01e89788 100644
--- a/obexd/src/obex.service.in
+++ b/obexd/src/obex.service.in
@@ -1,11 +1,16 @@
 [Unit]
 Description=Bluetooth OBEX service
+Requires=bluetooth.service
+After=bluetooth.service
 
 [Service]
 Environment="MAP_ROOT="map-messages""
 Type=dbus
 BusName=org.bluez.obex
-ExecStart=@pkglibexecdir@/obexd
+ExecStartPre=/bin/mkdir -p /var/bluetooth
+ExecStart=@pkglibexecdir@/obexd -s -a -r /var/bluetooth
+Restart=on-failure
 
 [Install]
+WantedBy=bluetooth.target
 Alias=dbus-org.bluez.obex.service
-- 
2.34.1

